core/src/main/java/org/testcontainers/utility/PreconditionUtils.java

```diff
@@ -0,0 +1,32 @@
+package org.testcontainers.utility;
+
+import lombok.experimental.UtilityClass;
+import org.testcontainers.containers.ContainerLaunchException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@UtilityClass
+public class PreconditionUtils {
```

Comment:
I think I'd prefer that this not be public scope - it's adding to our public API for something that is very specifically coupled to Docker Compose.

Reply:
I suppose i can make a helper class in DockerComposeContainer file and move all precondition's logic there

Reply:
I think that sounds sensible - it will make an already long class longer, but we don't have much in the way of options. Maybe Java modules will help with this one day :)